### ISSUE TITLE
Windows is not working

### ISSUE THREAD

**Issue by Bruninadm2** [hakuna-m/wubiuefi#302]
After installed Ubuntu budgie on wubiuefi windows 7 is not booting any more when I start on windows the pc reboot

**Comment by hakuna-m** [comment]
![](https://github.com/hakuna-m/wubiuefi/wiki/images/Install-Restart.png)

Do you mean that reboot ?

It should be only an one-time boot to install Ubuntu. If it doesn't work, you can choose Windows from Windows Boot Manager. If you use Windows 7 with UEFI (not default, but possible), see [here](https://github.com/hakuna-m/wubiuefi/wiki/FAQ#where-is-my-ubuntu-entry-in-windows-boot-manager-for-uefi-)

Note: It is not recommended to change BIOS mode from UEFI to Legacy/CSM or the other way round because the BIOS mode should always match that one of the installed Windows. see [here](https://github.com/hakuna-m/wubiuefi/wiki/FAQ#how-can-i-change-uefi-firmware-settings-from-windows-)

**Comment by Bruninadm2** [comment]
No. I boot the Ubuntu and works but when I boot to windows 7 the pc reboots automatically

**Comment by Bruninadm2** [comment]
> ![](https://github.com/hakuna-m/wubiuefi/wiki/images/Install-Restart.png)
> 
> Do you mean that reboot ?
> 
> It should be only an one-time boot to install Ubuntu. If it doesn't work, you can choose Windows from Windows Boot Manager. If you use Windows 7 with UEFI (not default, but possible), see [here](https://github.com/hakuna-m/wubiuefi/wiki/FAQ#where-is-my-ubuntu-entry-in-windows-boot-manager-for-uefi-)
> 
> Note: It is not recommended to change BIOS mode from UEFI to Legacy/CSM or the other way round because the BIOS mode should always match that one of the installed Windows. see [here](https://github.com/hakuna-m/wubiuefi/wiki/FAQ#how-can-i-change-uefi-firmware-settings-from-windows-)

No. The Linux boots normally but when I boot to windows 7 it restarts automatically

**Comment by hakuna-m** [comment]
Which menu did you use for booting to Windows ? If you use the GRUB menu, there is also the possibility to use Windows Boot Manager. 

For the Windows Boot Manager for BIOS press F8 during boot
For the Windows Boot Manager for UEFI see [here](https://github.com/hakuna-m/wubiuefi/wiki/FAQ#where-is-my-ubuntu-entry-in-windows-boot-manager-for-uefi-) 

Can you analyze it from Ubuntu [terminal](https://ubuntu.com/tutorials/command-line-for-beginners#3-opening-a-terminal):

- UEFI or BIOS ? 

`[ -d /sys/firmware/efi ] && echo UEFI || echo BIOS `
`cat /host/Users/*/AppData/Local/Temp/wubi*.log | grep "EFI boot"`

- GRUB menu entry for Windows:

`cat /boot/grub/grub.cfg | grep -i windows -A10`

**Comment by Bruninadm2** [comment]
> Which menu did you use for booting to Windows ? If you use the GRUB menu, there is also the possibility to use Windows Boot Manager.
> 
> For the Windows Boot Manager for BIOS press F8 during boot For the Windows Boot Manager for UEFI see [here](https://github.com/hakuna-m/wubiuefi/wiki/FAQ#where-is-my-ubuntu-entry-in-windows-boot-manager-for-uefi-)
> 
> Can you analyze it from Ubuntu [terminal](https://ubuntu.com/tutorials/command-line-for-beginners#3-opening-a-terminal):
> 
> * UEFI or BIOS ?
> 
> `[ -d /sys/firmware/efi ] && echo UEFI || echo BIOS ` `cat /host/Users/*/AppData/Local/Temp/wubi*.log | grep "EFI boot"`
> 
> * GRUB menu entry for Windows:
> 
> `cat /boot/grub/grub.cfg | grep -i windows -A10`

Windows boot manager

**Comment by hakuna-m** [comment]
> Windows boot manager

For UEFI or for BIOS ?

If Ubuntu Budgie works, it is helpful to check BIOS mode and the logs with Ubuntu Budgie.

You should find all logs with 

`ls /host/Users/*/AppData/Local/Temp/wubi*.log`

e.g. my latest log name is **/host/Users/hakuna-m/AppData/Local/Temp/wubi-20.04-rev344.log**

Probably, hakuna-m (Windows user name) and 20.04-rev344 (version number) are different in your case. The log should contain information about the used boot loader.

In general, in our tests the Wubiuefi installation has never changed a working Windows to a non working Windows.

One time, we deleted a menu entry for Windows (thinking it was redundant) which caused a non working Windows. But we did not use Wubiuefi to do so and it was not a huge problem because the used tool created a backup of the BCD file. The current BCD file was in folder **/boot/efi/EFI/Microsoft/Boot** and we could successfully replace that one with the backup.

As far as I know, in the past there was only one user who created here an issue because of a not working Windows after installation. In his case, the reason was a non standard boot loader for encrypted disks.

To put in a nutshell, maybe it is easily possible to replace the current not working Windows Boot Manager with a working one. But the information "Windows does not work" is not enough. There are also differences between Windows Boot Manager for BIOS and UEFI but no information which one is currently in use. Windows 7 can use both, default is the old one for BIOS.
